packages/db-model/src/model.ts

```diff
@@ -72,86 +70,96 @@ export function createBaseModel(resolver?: ConnectionResolver) {
-      return db.get<T>(this.getConnection(), this.table, params, trx);
-    }
+    public static findFirst<T>(params: object = {}, trx?: Knex.Transaction): Promise<T> {
+      return new Promise<T>(async (resolve, reject) => {
+        const [result] = await db.findFirst<T>(this.getConnection(), this.table, params, this.defaultOrderBy, trx);
 
-    /**
-     * Find record by it's id.
-     * Returns null if not found.
-     *
-     * @param {number} id
-     * @param {Knex.Transaction} trx
-     * @returns {Promise<T | null>}
-     */
-    public static getById<T>(id: number, trx?: Knex.Transaction): Promise<T | null> {
-      return db.getById<T>(this.getConnection(), this.table, id, trx);
+        resolve(result);
+      });
     }
 
     /**
-     * Finds a record based on the params.
+     * Find by primary key
      *
-     * @param {object} [params={}]
+     * @param {string} pk
      * @param {Knex.Transaction} trx
-     * @throws {ModelNotFoundError}
      * @returns {Promise<T>}
      */
-    public static async find<T>(params: object = {}, trx?: Knex.Transaction): Promise<T> {
-      const result = await this.get<T>(params, trx);
+    public static findByPk<T>(pk: string, trx?: Knex.Transaction): Promise<T> {
+      return new Promise<T>(async (resolve, reject) => {
+        const pkParams: { [key: string]: any } = {};
+        pkParams[this.pk] = pk;
 
-      if (!result) {
-        throw new ModelNotFoundError(this.name + ' not found');
-      }
+        const [result] = await db.findFirst<T>(this.getConnection(), this.table, pkParams, this.defaultOrderBy, trx);
 
-      return result;
+        resolve(result);
+      });
     }
 
     /**
-     * Find record by it's id.
-     * Throws an exception if not found.
+     * Finds records based on the params.
      *
-     * @param {number} id
+     * @param {object} [params={}]
      * @param {Knex.Transaction} trx
-     * @throws {ModelNotFoundError}
      * @returns {Promise<T>}
      */
-    public static findById<T>(id: number, trx?: Knex.Transaction): Promise<T> {
-      return this.find<T>({ id }, trx);
+    public static find<T>(params: object = {}, trx?: Knex.Transaction): Promise<T> {
+      return new Promise<T>(async (resolve, reject) => {
+        const result = await db.find<T>(this.getConnection(), this.table, params, this.defaultOrderBy, trx);
+
+        resolve(result);
+      });
     }
 
     /**
-     * Find all records based on the params.
+     * Finds records based on the params with records limit.
      *
      * @param {object} [params={}]
      * @param {Knex.Transaction} trx
-     * @returns {Promise<T[]>}
+     * @throws {ModelNotFoundError}
+     * @returns {Promise<T>}
      */
-    public static async findAll<T>(params: object = {}, trx?: Knex.Transaction): Promise<T[]> {
-      return db.findAll<T>(this.getConnection(), this.table, params, trx);
+    public static findWithPageAndSort<T>(
+      params: object = {},
```

Comment:
The `object` type won't work here.
```suggestion
      params: any = {},
```